Userland/Libraries/LibCore/MappedFile.h

```diff
@@ -7,36 +7,71 @@
 
 #pragma once
 
+#include <AK/ByteBuffer.h>
 #include <AK/Error.h>
+#include <AK/MemoryStream.h>
 #include <AK/Noncopyable.h>
 #include <AK/NonnullRefPtr.h>
 #include <AK/RefCounted.h>
+#include <AK/Stream.h>
 #include <LibCore/Forward.h>
 
 namespace Core {
 
-class MappedFile {
+class MappedFile : public SeekableStream {
```

Comment:
I tried :^) The problem is subtle: When you construct FixedMemoryStream, it determines whether it is read-only or read-write based on its parameter type (ReadonlyBytes vs. Bytes). However, when we are a child class, we have to call *one single overload* of the parent constructor, and which overload specifically is determined at compile time. So we can't provide this distinction, which would leave us with only one solution: pretend to FMS that we're always read-write and implement write protections ourselves by, again, overloading many stream functions. If we use composition instead, we can differentiate the constructor to use (see the m_stream initialization in the constructor) and let FMS handle write protections for us. In the end this also means that the class is smaller, since we don't need to store writability twice.

Reply:
What if we make the `FixedMemoryStream(Bytes)` constructor a `FixedMemoryStream(Bytes, bool writable = true)` constructor instead?